tutorials/truffle-dev-life-cycle.md

```diff
@@ -0,0 +1,538 @@
+npx truffle run moonbeam remove
+```
+
+## Deploying to Moonbase Alpha TestNet {: #deploying-to-moonbase-alpha }
+
+Now that we've been able to rapidly develop our contracts with our Moonbeam development node and feel confident with our code, we can move on to testing it on the Moonbase Alpha TestNet. 
+
+First, you'll need to update your `truffle-config.js` file and add in the private key of your account on Moonbase Alpha. The `privateKeyMoonbase` variable already exists, you just need to set it to your private key. **This is just for demonstration purposes only, never store your private keys in a JavaScript file**.
+
+Once you've set your account up, you can run your tests on Moonbase Alpha to make sure they work as expected on a live network:
+
+```
+npx truffle test --network moonbase
+```
+
+If your tests all pass, you can feel free to deploy your contracts using this command:
+
+```
+npx truffle migrate --network moonbase
+```
+
+Feel free to play around and interact with your contracts on the TestNet! Since DEV tokens have no real monetary value, now is a good time to work out any kinks before we deploy our contracts to Moonbeam MainNet where the tokens do have value!
+
+## Deploying to Production on Moonbeam MainNet {: #deploying-to-production }
+
+We thought we felt confident before, but now that we've tested our contracts on Moonbase Alpha we feel even more confident. So let's do it and test and deploy our contracts on Moonbeam!
+
+Again, you'll need to update your `truffle-config.js` and add in the private key of your account on Moonbeam. If you're interested in a secure way to add your private keys, you can check out the [Truffle Dashboard](https://trufflesuite.com/blog/introducing-truffle-dashboard/){target=_blank}, which allows you to connect to your MetaMask wallet without any configuration. For more information, please refer to the [Truffle docs on using the Truffle Dashboard](https://trufflesuite.com/docs/truffle/how-to/use-the-truffle-dashboard/){target=_blank}. Whatever you do, **never store your private keys in a JavaScript file**.
+
+Since the Moonbeam Truffle box doesn't come with the Moonbeam network configurations, you'll need to add them:
+
+```
+...
+module.exports = {
+  networks: {
+    ...
+    moonbeam: {
+      provider: () => {
+        ...
+        return new HDWalletProvider(
+          'PRIVATE-KEY-HERE',  // Insert your private key here
+          '{{ networks.moonbeam.rpc_url }}' // Insert your RPC URL here
+        )
+      },
+      network_id: {{ networks.moonbeam.chain_id }} (hex: {{ networks.moonbeam.hex_chain_id }}),
+    },
+  }
+}
+```
+
+If you're using Truffle Dashboard, you'll need to [add the host/port configuration for your dashboard](https://trufflesuite.com/docs/truffle/how-to/use-the-truffle-dashboard/#connecting-to-the-dashboard){target=_blank}.
+
+Once you've set your account up, you can run your tests on Moonbeam:
+
+```
+npx truffle test --network moonbeam
```

Comment:
This is not something that people actually do, so I would remove it for sure

Reply:
haha I would think that is maybe the most important place to run them 😅